src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -591,37 +425,58 @@ void TemporalMemory::compute(
                              activeColumnsUnsorted + activeColumnsSize);
   std::sort(activeColumns.begin(), activeColumns.end());
 
-  for (const ExcitedColumnData& excitedColumn : ExcitedColumns(activeColumns,
-                                                               activeSegments_,
-                                                               matchingSegments_,
-                                                               cellsPerColumn_))
+  const auto identity =
+    [](UInt c) { return c;};
```

Comment:
OK, sorry, better to put it this way, the code "looks" confusing/"hacky", a comment with "this is a higher-order function in C++11" would be imho just fine